src/wallet/wallet.cpp

```diff
@@ -4472,11 +4472,7 @@ void CWallet::LoadActiveScriptPubKeyMan(uint256 id, OutputType type, bool intern
 
 bool CWallet::IsLegacy() const
 {
-    if (m_internal_spk_managers.count(OutputType::LEGACY) == 0) {
-        return false;
-    }
-    auto spk_man = dynamic_cast<LegacyScriptPubKeyMan*>(m_internal_spk_managers.at(OutputType::LEGACY));
-    return spk_man != nullptr;
+    return (m_internal_spk_managers.count(OutputType::LEGACY) == 1);
```

Comment:
This change is simply incorrect. The original checked for `LegacyScriptPubKeyMan`. This now just checks for the presence of a `ScriptPubKeyMan` that provides legacy change addresses. That is completely not what this is for. `OutputType::LEGACY` is unrelated to `LegacyScriptPubKeyMan`, this could have used `OutputType::BECH32` and not been any different.

Reply:
Got it, now I understand the reason of the cast. But ATM all `m_internal_spk_managers` are the same `LegacyScriptPubKeyMan` instance right?

Reply:
> But ATM all `m_internal_spk_managers` are the same `LegacyScriptPubKeyMan` instance right?

Only for legacy wallets. For legacy wallets, there's only one `ScriptPubKeyMan` for all `spk_managers`.

However for descriptor wallets, there are multiple `DescriptorScriptPubKeyMan`s. This change is incorrect for descriptor wallets.